cmd/soroban-cli/src/commands/contract/fetch.rs

```diff
@@ -150,9 +158,19 @@ impl NetworkRunnable for Cmd {
         client
             .verify_network_passphrase(Some(&network.network_passphrase))
             .await?;
-        Ok(client.get_remote_wasm(&contract_id).await?)
+        let data_entry = client.get_contract_data(&contract_id).await?;
+        if let ScVal::ContractInstance(contract) = &data_entry.val {
+            return match &contract.executable {
+                ContractExecutable::Wasm(hash) => {
+                    Ok(client.get_remote_wasm_from_hash(&hash).await?)
```

Comment:
Instead of calling this function use the `get_ledger_entries` and assume `get_remote_wasm_...` is deprecated. The rpc-client should be a thin wrapper on the rpc api, and both the `get_remote_wasm_...` functions are logic that should live here.

Reply:
Makes sense. Perhaps we would want to revisit methods we expose in rpc-client in the future to make sure we don't expose methods we don't want to, before it gets bigger adoption and it's harder to remove/deprecate methods people are using?

Reply:
+1, we should do a pass over the lib before v22. The lib started by extracting logic from the cli, but we extracted a lot more than just the client and the boundary between two had a lot of cli logic on the rpc client side. We've moved a lot over, but there's still things there.